api/messaging_service.cc

```diff
@@ -46,7 +45,7 @@ future_json_function get_client_getter(sharded<netw::messaging_service>& ms, std
         using map_type = std::unordered_map<gms::inet_address, uint64_t>;
         auto get_shard_map = [f](messaging_service& ms) {
             std::unordered_map<gms::inet_address, unsigned long> map;
-            ms.foreach_client([&map, f] (const msg_addr& id, const shard_info& info) {
+            ms.foreach_client([&map, f] (const netw::msg_addr& id, const shard_info& info) {
```

Comment:
Add comment here on the commit message because github is stupid.

It looks like the next patch makes `std::operator<<` use `fmt::formatter` and not the other way around like the commit message implies. Also I do not understand why exactly `fmt::formatter` has the problem with the alias while the operator hadn't, but may be less intrusive way to fix it is to add another formatter for the alias?

Reply:
I think that the namespace problems were caused by me moving operator<< to namespace std instead of keeping it in the netw namespace, but I also encountered a similar issue with the finding std::hash when e.g. class gossiper had a `using msg_addr = netw::addr`.
I'll go over the commit message regarding the formatter to clarify it

Reply:
not sure.  I need to check if moving the hash to the netw namespace solves this problem.

Reply:
> But now you are moving it back to netw namespace, no? So may be the whole patch can be dropped?

dropped in v5